File: EJERCICIO 1/CONTROL_DE_HORA.cpp
#include <iostream>
#include <iomanip>
using namespace std;

int main() {
    int hora, minutos, segundos, opcion;

    cout << "Ingrese hora (0-23): ";
    cin >> hora;
    cout << "Ingrese minutos (0-59): ";
    cin >> minutos;
    cout << "Ingrese segundos (0-59): ";
    cin >> segundos;

    if(hora < 0 || hora > 23 || minutos < 0 || minutos > 59 || segundos < 0 || segundos > 59) {
        cout << "Hora invalida" << endl;
    } else {
        cout << "Hora registrada: " << setw(2) << setfill('0') << hora << ":"
             << setw(2) << setfill('0') << minutos << ":"
             << setw(2) << setfill('0') << segundos << endl;

        cout << "¿ Desea cambiarla? (1=Si, 0=No): ";
        cin >> opcion;

        if(opcion == 1) {
            cout << "Ingrese nueva hora (0-23): ";
            cin >> hora;
            cout << "Ingrese nuevos minutos (0-59): ";
            cin >> minutos;
            cout << "Ingrese nuevos segundos (0-59): ";
            cin >> segundos;

            if(hora < 0 || hora > 23 || minutos < 0 || minutos > 59 || segundos < 0 || segundos > 59) {
                cout << "Nueva hora invalida" << endl;
            } else {
                cout << "Nueva hora: " << setw(2) << setfill('0') << hora << ":"
                     << setw(2) << setfill('0') << minutos << ":"
                     << setw(2) << setfill('0') << segundos << endl;
            }
        }
    }
    return 0;
}

File: EJERCICIO 1/CONTROL_DE_HORA.java
//package APE2EJERCICIOS;

import java.util.Scanner;

public class CONTROL_DE_HORA {
    public static void main(String[] args) {
      

        Scanner sc = new Scanner(System.in);
        int hora, minutos, segundos, opcion;

        System.out.print("Ingrese hora (0-23): ");
        hora = sc.nextInt();
        System.out.print("Ingrese minutos (0-59): ");
        minutos = sc.nextInt();
        System.out.print("Ingrese segundos (0-59): ");
        segundos = sc.nextInt();

        if(hora < 0 || hora > 23 || minutos < 0 || minutos > 59 || segundos < 0 || segundos > 59) {
            System.out.println("Hora inválida");
        } else {
            System.out.printf("Hora registrada: %02d:%02d:%02d\n", hora, minutos, segundos);
            System.out.print("¿Desea cambiarla? (1=Sí, 0=No): ");
            opcion = sc.nextInt();

            if(opcion == 1) {
                System.out.print("Ingrese nueva hora (0-23): ");
                hora = sc.nextInt();
                System.out.print("Ingrese nuevos minutos (0-59): ");
                minutos = sc.nextInt();
                System.out.print("Ingrese nuevos segundos (0-59): ");
                segundos = sc.nextInt();

                if(hora < 0 || hora > 23 || minutos < 0 || minutos > 59 || segundos < 0 || segundos > 59) {
                    System.out.println("Nueva hora inválida");
                } else {
                    System.out.printf("Nueva hora: %02d:%02d:%02d\n", hora, minutos, segundos);
                }
            }
        }
        sc.close();
    }
}
